buildkite/job.go

```diff
@@ -176,19 +223,41 @@ func (j *Job) Run(agent *Agent) error {
 	// Store the process so we can cancel it later.
 	j.process = process
 
+	// Mark the build as started
+	j.StartedAt = time.Now().UTC().Format(time.RFC3339Nano)
+	_, err = agent.Client.JobUpdate(j)
+	if err != nil {
+		// We don't care if the HTTP request failed here. We hope that it
+		// starts working during the actual build.
+	}
+
 	// Start the process. This will block until it finishes.
 	err = process.Start()
 	if err == nil {
-		// Store the final output
-		j.Output = process.Output()
+		// Add the final output to the streamer
+		logStreamer.Process(process.Output())
 	} else {
-		j.Output = fmt.Sprintf("%s", err)
+		// Send the error as output
+		logStreamer.Process(fmt.Sprintf("%s", err))
 	}
 
 	// Mark the build as finished
 	j.FinishedAt = time.Now().UTC().Format(time.RFC3339Nano)
 	j.ExitStatus = j.process.ExitStatus
 
+	// Upload the header timings one last time
+	j.uploadHeaderTimes(agent)
+
+	// Stop the log streamer
+	logStreamer.Stop()
+
+	// Save how many chunks failed to upload
+	j.ChunksFailedCount = int(logStreamer.ChunksFailedCount)
+
+	if j.ChunksFailedCount > 0 {
+		logger.Warn("%d chunks failed to upload for this job", j.ChunksFailedCount)
```

Comment:
Yeah - it's probably a good idea. The storage location is tricky (windows would be different) and there's log rotation to worry about. Can you open an issue and I can take a look at doing this in another PR?